### PR TITLE
Synthetic Generation Choice Matters (Support/Industrial Selection)

### PR DESCRIPTION
**Pull request by Katskan**
# About the pull request

Allows both shipside and colony spawns to choose their generation, which has accompanying traits. 

Generation One is a minimally cosmetic, older model with more industrial construction and build, i.e. classic colony stats for health, movement, skills, etc with the exception of increased engineering skills compared to previous as they are industrial focused models. Older models still learn by doing, they're just slower. 

Generation Two is an updated cosmetic version of Generation One, they have the same physical and skill traits, but you can use custom appearance on them. 

Generation Three is meant to be a technical specialist, and is more fragile than Generation One or Two, but has better all-around support skills. 

# Explain why it's good for the game

Increases RP potential for characters who lean into their background in certain areas, such as an old rugged frontiersman, a heavy space construction worker, or a clinical aide. Also allows consistency between your ship and survivor character, which eliminates instances where someone may be forced to RP in a certain way because they are different on account of their spawn being different. 

# Testing Photographs and Procedure

Runs on local, tested both colony and ship spawns of each generation, all functioned as intended. 

# Changelog

:cl:
balance: enable generation selection to affect species and skillset for synthetics
balance: removed generation 3 leadership skills
balance: increased generation 1/2 engineering and construction skills
fix: fixed the Solaris PMC nightmare spawn being incorrect skillset
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/modules/gear_presets/synths.dm [hunk 1]
```diff
@@ -12,10 +11,22 @@
...
+			skills = /datum/skills/synthetic
+		if(SYNTH_GEN_TWO)
+			new_human.set_species(SYNTH_COLONY_GEN_TWO)
+			skills = /datum/skills/colonial_synthetic
+		if(SYNTH_GEN_ONE)
+			new_human.set_species(SYNTH_COLONY_GEN_ONE)
+			skills = /datum/skills/colonial_synthetic
+		else
+			new_human.set_species(SYNTH_GEN_THREE)
+			skills = /datum/skills/synthetic
```
Comment (harryob) [inline code/modules/gear_presets/synths.dm:29]: this is a singleton, we shouldn't really be altering the variables of it

why not just override /load_skills like

```dm
/datum/equipment_preset/synth/load_skills(mob/living/carbon/human/new_human, client/mob_client)
	new_human.allow_gun_usage = FALSE

	if(iscolonysynthetic(new_human) && !isworkingjoe(new_human))
		new_human.set_skills(/datum/skills/colonial_synthetic)
		return

	if(!mob_client)
		new_human.set_skills(/datum/skills/synthetic)
		return

	switch(mob_client.prefs.synthetic_type)
		if(SYNTH_GEN_ONE, SYNTH_GEN_TWO)
			new_human.set_skills(/datum/skills/colonial_synthetic)
		else
			new_human.set_skills(/datum/skills/synthetic)
```

you also won't need to double up code for /datum/equipment_preset/synth/survivor

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/gear_presets/synths.dm (hunk 1 above)
Comment (Katskan) [reply code/modules/gear_presets/synths.dm:29]: Awesome, thanks!

File: code/modules/gear_presets/synths.dm [hunk 2]
```diff
@@ -32,6 +39,22 @@
...
+
+    if(!mob_client)
+        new_human.set_skills(/datum/skills/synthetic)
+        return
+
+    switch(mob_client.prefs.synthetic_type)
+        if(SYNTH_GEN_ONE, SYNTH_GEN_TWO)
+            new_human.set_skills(/datum/skills/colonial_synthetic)
+        else
+            new_human.set_skills(/datum/skills/synthetic)
```
Comment (harryob) [inline code/modules/gear_presets/synths.dm:57]:
```suggestion
	new_human.allow_gun_usage = FALSE

	if(iscolonysynthetic(new_human) && !isworkingjoe(new_human))
		new_human.set_skills(/datum/skills/colonial_synthetic)
		return

	if(!mob_client)
		new_human.set_skills(/datum/skills/synthetic)
		return

	switch(mob_client.prefs.synthetic_type)
		if(SYNTH_GEN_ONE, SYNTH_GEN_TWO)
			new_human.set_skills(/datum/skills/colonial_synthetic)
		else
			new_human.set_skills(/datum/skills/synthetic)
```
spaces to tabs

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by harryob** [comment]
aaalso make sure to hit this when it's ready for review too, maints tend to ignore drafted prs 
CleanShot 2024-10-27 at 21 40 15@2x

**Review by harryob — APPROVED** [review]

**Pull request merged**